src/main/java/org/opentripplanner/middleware/utils/ItineraryUtils.java

```diff
@@ -0,0 +1,295 @@
+                    !equalsIgnoreCaseOrPreviousWasEmpty(
+                        previousItineraryLeg.routeLongName,
+                        newItineraryLeg.routeLongName
+                    ) ||
+                    !equalsIgnoreCaseOrPreviousWasEmpty(
+                        previousItineraryLeg.routeShortName,
+                        newItineraryLeg.routeShortName
+                    ) ||
+                    !equalsIgnoreCaseOrPreviousWasEmpty(
+                        previousItineraryLeg.headsign,
+                        newItineraryLeg.headsign
+                    ) ||
+                    !equalsOrPreviousWasNull(
+                        previousItineraryLeg.interlineWithPreviousLeg,
+                        newItineraryLeg.interlineWithPreviousLeg
+                    )
+            ) {
+                return false;
+            }
+
+            // make sure the transit trips are scheduled for the same time of the day
+            if (
+                !timeOfDayMatches(
+                    previousItineraryLeg.getScheduledStartTime(),
+                    newItineraryLeg.getScheduledStartTime()
+                ) || !timeOfDayMatches(
+                    previousItineraryLeg.getScheduledEndTime(),
+                    newItineraryLeg.getScheduledEndTime()
+                )
+            ) {
+                return false;
+            }
+        }
+
+        // if this point is reached, the itineraries are assumed to match
+        return true;
+    }
+
+    /**
+     * Checks whether two stops (OTP Places) match for the purposes of matching itineraries
+     */
+    private static boolean stopsMatch(Place stopA, Place stopB) {
+        // stop names must match
+        if (!StringUtils.equalsIgnoreCase(stopA.name, stopB.name)) return false;
+
+        // stop code must match
+        if (!equalsIgnoreCaseOrPreviousWasEmpty(stopA.stopCode, stopB.stopCode)) return false;
+
+        // stop positions must be no further than 5 meters apart
+        if (
+            DistanceUtils.radians2Dist(
+                DistanceUtils.distHaversineRAD(stopA.lat, stopA.lon, stopB.lat, stopB.lon),
+                DistanceUtils.EARTH_MEAN_RADIUS_KM
+            ) * 1000 > 5
+        ) {
+            return false;
+        };
+
+        // if this point is reached, the stops are assumed to match
+        return true;
+    }
+
+    /**
+     * Returns true if previous is null. Otherwise, returns Objects.equals.
+     */
+    private static boolean equalsOrPreviousWasNull (Object previous, Object newer) {
+        return previous == null || Objects.equals(previous, newer);
+    }
+
+    /**
+     * Returns true if the previous string was empty. Otherwise, returns if the strings are equal ignoring case.
+     */
+    private static boolean equalsIgnoreCaseOrPreviousWasEmpty(String previous, String newer) {
+        return StringUtils.isEmpty(previous) || StringUtils.equalsIgnoreCase(previous, newer);
```

Comment:
How come a third party lib is used to compare strings?

Reply:
Used String.equalsIgnoreCase in 52d438aad1badb194c212a736a16d72aff3da526